addons/fleet/fleet.py

```diff
@@ -1,818 +1,725 @@
-            }
-        }
+class FleetVehicleModel(models.Model):
 
     _name = 'fleet.vehicle.model'
     _description = 'Model of a vehicle'
-    _order = 'name asc'
-
-    _columns = {
-        'name': fields.char('Model name', required=True),
-        'brand_id': fields.many2one('fleet.vehicle.model.brand', 'Make', required=True, help='Make of the vehicle'),
-        'vendors': fields.many2many('res.partner', 'fleet_vehicle_model_vendors', 'model_id', 'partner_id', string='Vendors'),
-        'image': fields.related('brand_id', 'image', type="binary", string="Logo"),
-        'image_medium': fields.related('brand_id', 'image_medium', type="binary", string="Logo (medium)"),
-        'image_small': fields.related('brand_id', 'image_small', type="binary", string="Logo (small)"),
-    }
-
-
-class fleet_vehicle_model_brand(osv.Model):
-    _name = 'fleet.vehicle.model.brand'
-    _description = 'Brand model of the vehicle'
-
-    _order = 'name asc'
-
-    def _get_image(self, cr, uid, ids, name, args, context=None):
-        result = dict.fromkeys(ids, False)
-        for obj in self.browse(cr, uid, ids, context=context):
-            result[obj.id] = tools.image_get_resized_images(obj.image)
-        return result
-
-    def _set_image(self, cr, uid, id, name, value, args, context=None):
-        return self.write(cr, uid, [id], {'image': tools.image_resize_image_big(value)}, context=context)
-
-    _columns = {
-        'name': fields.char('Make', required=True),
-        'image': fields.binary("Logo",
-            help="This field holds the image used as logo for the brand, limited to 1024x1024px."),
-        'image_medium': fields.function(_get_image, fnct_inv=_set_image,
-            string="Medium-sized photo", type="binary", multi="_get_image",
-            store = {
-                'fleet.vehicle.model.brand': (lambda self, cr, uid, ids, c={}: ids, ['image'], 10),
-            },
-            help="Medium-sized logo of the brand. It is automatically "\
-                 "resized as a 128x128px image, with aspect ratio preserved. "\
-                 "Use this field in form views or some kanban views."),
-        'image_small': fields.function(_get_image, fnct_inv=_set_image,
-            string="Smal-sized photo", type="binary", multi="_get_image",
-            store = {
-                'fleet.vehicle.model.brand': (lambda self, cr, uid, ids, c={}: ids, ['image'], 10),
-            },
-            help="Small-sized photo of the brand. It is automatically "\
-                 "resized as a 64x64px image, with aspect ratio preserved. "\
-                 "Use this field anywhere a small image is required."),
-    }
-
-
-class fleet_vehicle(osv.Model):
+    _order = 'name asc, id asc'
+
+    @api.multi
+    def name_get(self):
+        return self.mapped(lambda m: (m.id, '/'.join(filter(None, (m.make_id.name, m.name)))))
+
+    @api.onchange('make_id')
+    def on_change_make(self):
+        self.image = self.make_id.image
```

Comment:
Please check whether this is necessary. The field `image` is a related field, and onchanges are automatic on those fields.